pkg/mutation/match.go

```diff
@@ -59,57 +60,61 @@ func Matches(match mutationsv1.Match, obj runtime.Object, ns *corev1.Namespace)
 		return false, nil
 	}
 
+	clusterScoped := ns == nil || obj.GetObjectKind().GroupVersionKind().GroupKind() == schema.GroupKind{Group: "", Kind: "Namespace"}
+
 	if match.Scope == apiextensionsv1beta1.ClusterScoped &&
-		meta.GetNamespace() != "" {
+		!clusterScoped {
 		return false, nil
 	}
 
 	if match.Scope == apiextensionsv1beta1.NamespaceScoped &&
-		meta.GetNamespace() == "" {
+		clusterScoped {
 		return false, nil
 	}
 
-	found := false
-	for _, n := range match.Namespaces {
-		if meta.GetNamespace() == n {
-			found = true
-			break
+	if ns != nil {
```

Comment:
How do we treat `match.Namespaces` and `match.ExcludedNamespaces` when object is a  Namespace resource?

Reply:
Part of the calling conventions is to populate ns if it can be populated (namespaced resource, namespace itself).

We are already doing this for the webhook:

https://github.com/open-policy-agent/gatekeeper/blob/db5a58054ceb07a1a63d951e2bacd9b21aad9076/pkg/webhook/mutation.go#L171-L196

but I'd probably be good to document this requirement in the function doc comment above.

the `clusterScoped` var assignment adds checking if the object is a namespace b/c it's the one cluster scoped resource where ns != nil.

Maybe we should test to make sure that if Kind == Namespace, ns != nil and raise an error if we get the wrong input?

Reply:
ah ok I forgot that was checked in the webhook.

> Maybe we should test to make sure that if Kind == Namespace, ns != nil and raise an error if we get the wrong input?

yea I think thats a good defensive thing to do